server/mldb_server.cc

```diff
@@ -389,6 +390,17 @@ shutdown()
 
     // Graphite logging: just log a message bracketing service shutdown
     recordHit("serviceStopped");
+
+    int tasksBefore = -1;
+    int tasks;
+    while ((tasks = BackgroundTaskBase::getPendingTasks()) != 0) {
+        if (tasks != tasksBefore) {
+            tasksBefore = tasks;
+            INFO_MSG(logger) << "Still waiting for " << tasks
+                             << " task(s)";
+        }
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+    }
```

Comment:
On hold. This is a band aid. Shutdown should shutdown and join services, thread pools and the like properly.